### PR TITLE
feat: Refactor to use a new atClient class

### PR DESCRIPTION
**Pull request by cpswan**
Backporting work from the Infrafon implementation so that we have a more generic way of using atProtocol features.

**- What I did**

* Moved source files to `src`
* Added `build.py` to create .mpy files (necessary to keep enough RAM on the Pico W)
* Made multi platform, so it can be tested on Linux

**- How to verify it**

Try out the Linux version first, then build the Pi version

**TODO**

Reimplement the REPL.

**- Description for the changelog**

feat: Refactor to use a new atClient class

### CODE REVIEW

**Comment by cpswan** [comment]
This is still very much a work in progress, but it's got as far as being able to send a message to a modified version of atTalk.

Demo to follow next week...

File: src/version.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+major = '0'
+minor='0'
+build='alvaro'
+date='2023-06-13 21:05:44'
```
Comment (xavierchanth) [inline src/version.py]: Is this a leak of local config into the repo?

**Review by xavierchanth — DISMISSED** [review]
Just 1 Q, otherwise LGTM

File: src/version.py (hunk 1 above)
Comment (realvarx) [reply src/version.py]: That is normal, it changes automatically when `build.py` is executed

Comment (xavierchanth) [reply src/version.py]: In that case, we should add it to the .gitignore... unless there is some reason not to that I'm unaware of?

Comment (realvarx) [reply src/version.py]: Yes, also IMO that should be in `.gitignore` I didn't add it because @cpswan included it

**Review by realvarx — APPROVED** [review]
LGTM

**Pull request merged**